master/buildbot/process/build.py

```diff
@@ -291,8 +283,59 @@ def startBuild(self, build_status, workerforbuilder):
             log.err(Failure())
             self.buildFinished(['Build.setupBuild', 'failed'], EXCEPTION)
             return
+
         # flush properties in the beginning of the build
         yield self._flushProperties(None)
+
+        try:
+            ready = yield workerforbuilder.prepare(self)
+        except Exception:
+            log.err(Failure(), 'while preparing workerforbuilder:')
+            self.buildFinished(["worker", "not", "available"], RETRY)
```

Comment:
note the status text is actually ignored. that text will never be displayed (perhaps, we should fix that in self.buildFinished)